### PR TITLE
WIP: ENH: UI Listbox

### PR DESCRIPTION
**Pull request by MarcCote**
I've refactored @ranveeraggarwal's code for the `FileSelectMenu2D` and made a ListBox that should be more flexible.

This new component supports multiselection via ctrl+click and shift+click. Also, the example in `viz_ui.py` shows how to hook a callback function such that it is called whenever the list selection changes.

What's missing is making `FileSelectMenu2D` use `ListBox2D`.

Here's a small demo:
listbox
**EDIT:** Updated demo.

### CODE REVIEW

File: dipy/viz/ui.py [hunk 1]
```diff
@@ -1907,6 +1895,317 @@ def handle_events(self, actor):
...
+        self.view_offset = 0
+        self._build()
+        self.set_center(position)
+        self.update()
+
+        # Define events captured by this UI component.
+        self.add_callback(self.up_button.actor, "LeftButtonPressEvent",
+                          self.up_button_callback)
+        self.add_callback(self.down_button.actor, "LeftButtonPressEvent",
+                          self.down_button_callback)
```
Comment (ranveeraggarwal) [inline dipy/viz/ui.py]: In other UI elements I have separated the event handling logic in a separate function. This is just to keep the constructor a little clean.

**Comment by ranveeraggarwal** [comment]
This is awesome! We can now remove `FileSelectMenuText2D` and modify `FileSelectMenu2D` to use this - once this is merged.

**Comment by ranveeraggarwal** [comment]
On a further thought, `FileSelectMenu2D` is just a very specific use case of the `ListBox2D`, so we can remove it entirely and build a `FileMenu` element (a composite element) that uses the `ListBox2D` element with functions of the existing `FileSelectMenu2D`.

**Comment by MarcCote** [comment]
Waiting after PR #1385 and PR #1362

**Comment by MarcCote** [comment]
Rebased on top of #1441.

**Comment by MarcCote** [comment]
I've updated the demo gif (see the first post).

**Comment by skoudoro** [comment]
Let me test it quickly on windows @ranveeraggarwal. I will do it right now

**Comment by skoudoro** [comment]
and should we merge #1441 first?

**Comment by ranveeraggarwal** [comment]
Yeah, that makes sense @skoudoro

**Comment by skoudoro** [comment]
After playing with this UI, it works like a charm,  +1 for merging when it is possible

**Comment by dmreagan** [comment]
I'm having some hardware problems, so I'm not sure when I'll be able to look at this. If you all approve, go ahead and merge without me.

File: dipy/viz/ui.py (hunk 1 above)
Comment (MarcCote) [reply dipy/viz/ui.py]: Good point.

**Comment by Garyfallidis** [comment]
Needs rebase and update @MarcCote !

**Comment by Garyfallidis** [comment]
@MarcCote can you fix the pep8 issues?

**Comment by karandeepSJ** [comment]
I have added a scroll bar to this. 
scroll

**Comment by dmreagan** [comment]
thanks @MarcCote now let's put back the FileSelectMenu

**Pull request merged**